ingestion/justice_data_source/source.py

```diff
@@ -54,17 +66,68 @@ def create(cls, config_dict, ctx):
         return cls(ctx, config)
 
     def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
-        # TODO generate metadata for the dashboard itself
+        all_chart_data = self.client.list_all(self.config.exclude_id_list)
+
+        # create group entities for publication owners
+        owner_emails = {chart["owner_email"] for chart in all_chart_data}
+        for owner_email in owner_emails:
+            mce = self._make_group(owner_email)
+            wu = MetadataWorkUnit("single_mce", mce=mce)
+            self.report.report_workunit(wu)
+            yield wu
 
-        for chart_data in self.client.list_all():
+        # creates each chart entity
+        for chart_data in all_chart_data:
             mce = self._make_chart(chart_data)
             wu = MetadataWorkUnit("single_mce", mce=mce)
             self.report.report_workunit(wu)
             yield wu
 
+        # adds domains to each chart created previously (if chart has a domain)
+        for chart_data in all_chart_data:
```

Comment:
Not essential, but I was wondering if this can be combined with `_make_chart` above as they both modify the chart entity. That way you would half the number of metadata work units.

Reply:
This is difficult because you cannot add domain as part of a ChartSnapshotClass - it needs to be a MetadataChangeProposalEvent rather than being included in the MetadataChangeEvent